include/irr/asset/filters/dithering/CWhiteNoiseDither.h

```diff
@@ -29,12 +32,28 @@ namespace irr
 
 				static float get(const state_type* state, const core::vectorSIMDu32& pixelCoord, const int32_t& channel)
 				{
-					// TODO: to define in future
+					auto getWangHash = [&]()
+					{
+						IImage::SBufferCopy::getLocalByteOffset(pixelCoord, bufferStridesHash);
+						uint32_t seed = IImage::SBufferCopy::getLocalByteOffset(pixelCoord, bufferStridesHash) * channel;
+
+						seed = (seed ^ 61) ^ (seed >> 16);
+						seed *= 9;
+						seed = seed ^ (seed >> 4);
+						seed *= 0x27d4eb2d;
+						seed = seed ^ (seed >> 15);
+						return seed;
+					};
+					
+					const auto hash = static_cast<double>(getWangHash());
+					return static_cast<float>(static_cast<double>(hash) / double(~0u));
```

Comment:
your wang hash is done in 32bit uint, your return type is `float`... there's no need for using `double`s

Reply:
Okay, do i will devide by float and stop using those casts

Reply:
corrected in following commit